File: ExcludedSamples/Reports/C1Report/VB/C1dView/C1dView.ReportSelectDialog.vb
Option Explicit On
Option Strict On

Imports System
Imports System.ComponentModel
Imports C1.C1Preview
Imports C1.C1Report
Imports C1.Win.C1Preview
Imports C1.Win.C1Ribbon
Imports C1.Win.C1Command

Public Class ReportSelectDialog

    Private _fileName As String = Nothing

    Public Sub New()
        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        AddHandler lstReports.SelectedIndexChanged,
            Sub(s, e)
                If (lstReports.SelectedItems.Count > 0) Then
                    ReportName = lstReports.SelectedItems(0).Text
                Else
                    ReportName = String.Empty
                End If
            End Sub

        AddHandler lstReports.DoubleClick, Sub(s, e) btnOk.PerformClick()

        AddHandler btnOk.Click,
            Sub(s, e)
                Me.DialogResult = System.Windows.Forms.DialogResult.OK
                Me.Close()
            End Sub

        AddHandler btnCancel.Click,
            Sub(s, e)

                Me.DialogResult = System.Windows.Forms.DialogResult.Cancel
                Me.Close()
            End Sub
    End Sub

    Public Property ReportName As String

    Public ReadOnly Property HasReports As Boolean
        Get
            Return lstReports.Items.Count > 0
        End Get
    End Property

    Public Property FileName As String
        Get
            Return _fileName
        End Get
        Set(value As String)
            lstReports.Clear()
            Try
                Dim reports = C1Report.GetReportList(value)
                For Each repName As String In reports
                    lstReports.Items.Add(repName)
                Next
                If (lstReports.Items.Count > 0) Then
                    lstReports.SelectedIndices.Add(0)
                End If
                _fileName = value
            Catch

                _fileName = Nothing
            End Try
        End Set
    End Property
End Class

File: ExcludedSamples/NetFramework/Reports/C1Report/Cs/C1dView/C1dView.ReportSelectDialog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using C1.C1Report;

namespace C1dView
{
    public partial class ReportSelectDialog : Form
    {
        private string _fileName = null;

        public ReportSelectDialog()
        {
            InitializeComponent();
            lstReports.SelectedIndexChanged += (s, e) =>
                {
                    if (lstReports.SelectedItems.Count > 0)
                        ReportName = lstReports.SelectedItems[0].Text;
                    else
                        ReportName = string.Empty;
                };
            lstReports.DoubleClick += (s, e) =>
                {
                    btnOk.PerformClick();
                };
            btnOk.Click += (s, e) =>
                {
                    this.DialogResult = System.Windows.Forms.DialogResult.OK;
                    this.Close();
                };
            btnCancel.Click += (s, e) =>
            {
                this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
                this.Close();
            };
        }

        public string ReportName { get; set; }

        public bool HasReports { get { return lstReports.Items.Count > 0; } }

        public string FileName
        {
            get { return _fileName; }
            set
            {
                lstReports.Clear();
                try
                {
                    var reports = C1Report.GetReportList(value);
                    foreach (string repName in reports)
                        lstReports.Items.Add(repName);
                    if (lstReports.Items.Count > 0)
                        lstReports.SelectedIndices.Add(0);
                    _fileName = value;
                }
                catch
                {
                    _fileName = null;
                }
            }
        }
    }
}
